index-handlers/src/main/java/no/sikt/nva/nvi/index/aws/CandidateQuery.java

```diff
@@ -0,0 +1,313 @@
+    }
+
+    private static Query createMatchAllQuery() {
+        return QueryBuilders.matchAll().build()._toQuery();
+    }
+
+    private static Query multipleApprovalsQuery() {
+        return mustMatch(rangeFromQuery(NUMBER_OF_APPROVALS, MULTIPLE));
+    }
+
+    private static Query nestedQuery(String path, Query... queries) {
+        return new NestedQuery.Builder()
+                   .path(path)
+                   .query(mustMatch(queries))
+                   .build()._toQuery();
+    }
+
+    private static Query statusQuery(String customer, ApprovalStatus status) {
+        return nestedQuery(APPROVALS,
+                           termQuery(customer, jsonPathOf(APPROVALS, ID)),
+                           termQuery(status.getValue(), jsonPathOf(APPROVALS, APPROVAL_STATUS)));
+    }
+
+    private static Query termQuery(String value, String field) {
+        return new TermQuery.Builder()
+                   .value(new FieldValue.Builder().stringValue(value).build())
+                   .field(field)
+                   .build()._toQuery();
+    }
+
+    private static String jsonPathOf(String... args) {
+        return String.join(JSON_PATH_DELIMITER, args);
+    }
+
+    private static Query rangeFromQuery(String field, int greaterThanOrEqualTo) {
+        return new RangeQuery.Builder().field(field).gte(JsonData.of(greaterThanOrEqualTo)).build()._toQuery();
+    }
+
+    private static Query containsPendingStatusQuery() {
+        return nestedQuery(APPROVALS, termQuery(PENDING.getValue(), jsonPathOf(APPROVALS, APPROVAL_STATUS)));
+    }
+
+    private static Query assignmentsQuery(String username, String customer) {
+        return nestedQuery(APPROVALS,
+                           termQuery(customer, jsonPathOf(APPROVALS, ID)),
+                           termQuery(username, jsonPathOf(APPROVALS, ASSIGNEE)));
+    }
+
+    private static Query termsQuery(List<String> values, String field) {
+        var termsFields = values.stream().map(FieldValue::of).toList();
+        return new TermsQuery.Builder()
+                   .field(field)
+                   .terms(new TermsQueryField.Builder().value(termsFields).build())
+                   .build()._toQuery();
+    }
+
+    private static Query matchQuery(String value, String field) {
+        return new MatchQuery.Builder().field(field)
+                   .query(new FieldValue.Builder().stringValue(value).build())
+                   .build()._toQuery();
+    }
+
+    private static Query contributorQuery(List<String> institutions) {
+        return nestedQuery(jsonPathOf(PUBLICATION_DETAILS, CONTRIBUTORS),
+                           QueryBuilders.bool().must(
+                               matchAtLeastOne(
+                                   termsQuery(institutions, jsonPathOf(PUBLICATION_DETAILS, CONTRIBUTORS, AFFILIATIONS, ID)),
+                                   termsQuery(institutions, jsonPathOf(PUBLICATION_DETAILS, CONTRIBUTORS, AFFILIATIONS, PART_OF))
```

Comment:
Loooooooong long lines (if you get the reference, cool, if not…meh…the lines are long and codacy is upset)